### PR TITLE
Add an argument for disabling unused controllers

### LINKED ISSUE #612: Make it possible to turn off fluent-bit/fluentd controllers when it is not used
### Is your feature request related to a problem? Please describe.

Currently, both controllers (`fluentd` and `fluent-bit`) work together and there is no option to disable one of them even when we do not want to use it. This adds the requirement to install all of the CRDs and to create an RBAC with all of the roles there.

### Describe the solution you'd like

The installation of the fluent operator should have the option to disable one of the controllers.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kristian-ZH**
### What this PR does / why we need it:
This PR adds a new flag called `disable-component-controllers` which disabled the given component's controllers
The possible values are: `fluent-bit` and `fluentd`
The default value of the argument will be an empty string which will not break the current work of the operator and will not break its consumers.

The benefit which this PR brings is that the operator's consumers will no any more have to install CRDs and expand the operator's RBAC with configurations for a component that they do not use in their setups.

### Which issue(s) this PR fixes:

Fixes #612

### Does this PR introduced a user-facing change?

```release-note
A new argument called disable-component-controllers is added which can turn off the unused component's controllers
```

### CODE REVIEW

**Review by wenchajun — DISMISSED** [review]

**Comment by wanjunlei** [comment]
It should not add the fluentbit CRDs to scheme when fluent-bit is disabled.

https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-manager/main.go#L53

File: cmd/fluent-manager/main.go [hunk 1]
```diff
@@ -42,6 +44,11 @@ import (
 	// +kubebuilder:scaffold:imports
 )
 
+const (
+	fluentBitComponentName = "fluent-bit"
```
Comment (benjaminhuo) [inline cmd/fluent-manager/main.go:48]:
```suggestion
	fluentBitName = "fluent-bit"
```

File: cmd/fluent-manager/main.go [hunk 2]
```diff
@@ -42,6 +44,11 @@ import (
...
+	fluentDComponentName   = "fluentd"
```
Comment (benjaminhuo) [inline cmd/fluent-manager/main.go:49]:
```suggestion
	fluentdName   = "fluentd"
```

File: cmd/fluent-manager/main.go [hunk 3]
```diff
@@ -61,9 +68,11 @@ func main() {
 	var probeAddr string
 	var watchNamespaces string
 	var logPath string
+	var disableComponentControllers string
```
Comment (benjaminhuo) [inline cmd/fluent-manager/main.go:71]:
```suggestion
	var disabledControllers string
```

File: cmd/fluent-manager/main.go [hunk 4]
```diff
@@ -61,9 +68,11 @@ func main() {
...
 	flag.StringVar(&watchNamespaces, "watch-namespaces", "", "Optional comma separated list of namespaces to watch for resources in. Defaults to cluster scope.")
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.StringVar(&disableComponentControllers, "disable-component-controllers", "", "Optional argument which accepts two values: fluent-bit and fluentd. It will not start the controllers related to the given component.")
```
Comment (benjaminhuo) [inline cmd/fluent-manager/main.go:75]:
```suggestion
	flag.StringVar(&disabledControllers, "disable-controllers", "", "Optional argument that accepts two values: fluent-bit and fluentd. The specific controllers will not be started if they're disabled")
```

File: cmd/fluent-manager/main.go [hunk 5]
```diff
@@ -103,54 +112,69 @@ func main() {
...
 	}
 
-	if err = (&controllers.FluentBitConfigReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("FluentBitConfig"),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "FluentBitConfig")
-		os.Exit(1)
+	fluentBitEnabled, fluentDEnabled := true, true
```
Comment (benjaminhuo) [inline cmd/fluent-manager/main.go:115]:
```suggestion
	fluentBitEnabled, fluentdEnabled := true, true
```

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
> https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-manager/main.go#L53

The init runs before the main function in which the disable flag is parsed, so how to do this actually?
And the helm will still try to install the CRDs even if the controller is disabled

**Comment by benjaminhuo** [comment]
> It should not add the fluentbit CRDs to scheme when fluent-bit is disabled.
> 
> https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-manager/main.go#L53
@wanjunlei 
The init runs before the main function in which the disable flag is parsed, so how to do this actually?
And the helm will still try to install the CRDs even if the controller is disabled

**Comment by wanjunlei** [comment]
> > It should not add the fluentbit CRDs to scheme when fluent-bit is disabled.
> > https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-manager/main.go#L53
> 
> @wanjunlei The init runs before the main function in which the disable flag is parsed, so how to do this actually? And the helm will still try to install the CRDs even if the controller is disabled

Just move these codes to the main. 
The helm chart should be modified accordingly to support not installing the fluentbit CRDS.

**Comment by Kristian-ZH** [comment]
All the comments should be resolved, can you please take a second look?

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@Kristian-ZH Nice move! I like the idea of moving the crds to subchart!
Thanks!

**Comment by benjaminhuo** [comment]
@Kristian-ZH I'm going to invite you as the fluent-operator maintainer!

**Comment by benjaminhuo** [comment]
@Kristian-ZH would you help to revise the install guide? https://github.com/fluent/fluent-operator#deploy-fluent-operator-with-helm

**Pull request merged**